### DISCUSSION TITLE
Bug: Installing and Running without Docker don't works anymore

### DISCUSSION THREAD

**Discussion by MichaelFomenko** [open-webui/open-webui#2053] [General]
# Bug Report

## Description

**Bug Summary:**
it don't work anymore, it asks for the build directory, but there is none. If I call http://localhost:8080/ it gives me only an empty Json file.
OS: Ubuntu 22.04

**Steps to Reproduce:**
```
Build and Install 🛠️
Run the following commands to install:

git clone https://github.com/open-webui/open-webui.git
cd open-webui/

# Copying required .env file
cp -RPp .env.example .env

# Building Frontend Using Node
npm i
npm run build

# Serving Frontend with the Backend
cd ./backend
pip install -r requirements.txt -U
bash start.sh
```

**Comment by MichaelFomenko** [comment]
I found out, that the npm i have problems, it tells that 
```
npm ERR! code 1
npm ERR! path /home/user/open-webui/node_modules/cypress
npm ERR! command failed
npm ERR! command sh -c node index.js --exec install
```

**Reply by tjbck** [reply]
Could you try clean installing?

**Reply by MichaelFomenko** [reply]
Can you reproduce this problem on your system?

**Reply by tjbck** [reply]
Not at all 😅 Our dev env uses the same setup and it works flawlessly on all three of my machines (Mac, Windows, Ubuntu)

**Reply by MichaelFomenko** [reply]
Which Node version do you use on Ubuntu? I tryed different versions. The "sudo snap install node --classic" its the Node Version: 20.12.2 worked without any problems until Open Web UI v0.1.122 after this version it works not any more.

**Reply by tjbck** [reply]
Strange, could you try `npm ci` and see if that fixes your issue?

**Reply by MichaelFomenko** [reply]
when I run npm ci:
```
npm error code 1
npm error path /home/user/open-webui/node_modules/cypress
npm error command failed
npm error command sh -c node index.js --exec install
```

**Reply by MichaelFomenko** [reply]
and I tryed the same thing with an different Ubuntu 22.04 install on my other machine and it have the same problem with cypress

**Reply by tjbck** [reply]
Interesting, no issue with `v20.13.0` on ubuntu here.

See if one of the solutions here: https://stackoverflow.com/questions/67399785/how-to-solve-npm-install-error-npm-err-code-1 does anything for you.

**Reply by MichaelFomenko** [reply]
why do you have the Version v20.13.0 on Ubuntu 22.04? On Ubuntu 22.04 the newest Snap version is: v20.12.2

**Reply by tjbck** [reply]
Installed with https://nodejs.org/en/download/package-manager

**Reply by MichaelFomenko** [reply]
why do you use not standard Installation process? please use "sudo snap install node --classic" on the dev env because everybody uses the standard installation on Ubuntu.

**Reply by justinh-rahb** [reply]
> why do you use not standard Installation process? please use "sudo snap install node --classic" on the dev env because everybody uses the standard installation on Ubuntu.

Telling a maintainer how to maintain their development environment is unlikely to go very far.

**Reply by tjbck** [reply]
My dev env is macos apple silicon btw 😅 Only using ubuntu for testing/deployment!

**Reply by MichaelFomenko** [reply]
ok, if you not using the standard Installation procedure, that not containing the same version than in the users Environment. Please tell in the Installation Dokumentation that the user should install the Package in this way and with the concrete Version number. Because how should the user know this?

**Reply by tjbck** [reply]
Have you tried `npm cache clean --force` ?

**Reply by tjbck** [reply]
`sudo snap install node --classic` downloads 20.13.0 for me, trying to check if `v20.12.0` is causing the issue, investigating atm.

**Reply by MichaelFomenko** [reply]
yes I tryed "npm cache clean --force" it changed nothing. And maybe your previous Installation of node is still there and this is why it shows you not the snap version but the manual installed. I am now trying to install Ubuntu 24.04 maybe there is no problems with that in the new Ubuntu version.

**Reply by tjbck** [reply]
Tested with `v20.12.2` works flawlessly. 

![image](https://github.com/open-webui/open-webui/assets/25473318/f12c7fcf-8a59-4b51-b15b-b172dda17c20)


https://nodejs.org/en/download/package-manager is the standard way to install nodejs by the way, not every linux distro has access to ubuntu snap store.

**Reply by MichaelFomenko** [reply]
I installed Node with your package-manager and now it works on Ubuntu 22.04. Like I told you, if you not using the Standard installation Procedure, tell the user in the Installation & Building Dokumentation how you installed a non standard Package. Because it solved now my problem and I would never think about installing node not in snap or apt.
